Header      header
Waypoint    waypoint
uint32      index      # index to insert at